### PR TITLE
Master

### PR DESCRIPTION
**Pull request by 1vn**
input_fn and some changes to deep_ae to make it compatible

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -99,3 +99,5 @@ ENV/
 
 # mypy
 .mypy_cache/
+
+/tmp
```
Comment (aidangomez) [inline .gitignore]: Shouldn't need to add this, Git won't be tracking anything outside of this project's directory.

File: data/mnist.py [hunk 2]
```diff
@@ -47,25 +47,26 @@ def _input_fn():
...
-            data_set.labels)
+            data_set.images)
 
     image, label = tf.train.slice_input_producer([input_images, input_labels])
-    return tf.train.batch([image, label], batch_size=params.batch_size)
+
+    imageBatch, labelBatch = tf.train.batch(
+        [image, label], batch_size=params.batch_size)
+
+    print(imageBatch.shape, labelBatch.shape)
```
Comment (aidangomez) [inline data/mnist.py]: Don't print this

File: data/mnist.py [hunk 3]
```diff
@@ -25,8 +28,7 @@ def _input_fn():
     # TODO(aidan): generalize image reading?
     # TODO(aidan): don't assume png or channels == 1
     decoded_images = [
-        tf.image.decode_png(
-            image, channels=1, dtype=tf.uint16)
+        tf.image.decode_png(image, channels=1, dtype=tf.uint16)
```
Comment (aidangomez) [inline data/mnist.py]: Install PyYapf and use the formatter, it will automatically format your code to match mine.

File: models/deep_ae.py [hunk 4]
```diff
@@ -13,6 +13,7 @@ def autoencoder(features, labels, mode):
       for ei in range(ed):
         tier_width = (
             (ed - ei - 1) * data_size + ei * params.hidden_size) // (ed - 1)
+
```
Comment (aidangomez) [inline models/deep_ae.py]: not necessary

File: models/deep_ae.py [hunk 5]
```diff
@@ -36,14 +37,15 @@ def autoencoder(features, labels, mode):
...
-                                              [params.batch_size, 28, 28, 1]))
+      tf.summary.image("reconstr", tf.reshape(x, [params.batch_size, 28, 28,
+                                                  1]))
       tf.summary.image("original",
                        tf.reshape(labels, [params.batch_size, 28, 28, 1]))
 
       mse_loss = tf.losses.mean_squared_error(labels, x)
       weight_reg = tf.add_n(
-          [tf.nn.l2_loss(x) for x in tf.trainable_variables()])
+          [tf.nn.l2_loss(l2l) for l2l in tf.trainable_variables()])
```
Comment (aidangomez) [inline models/deep_ae.py]: change `l2l` to `w`

File: models/deep_ae.py [hunk 6]
```diff
@@ -60,6 +62,8 @@ def autoencoder(features, labels, mode):
           optimizer=tf.train.AdamOptimizer(lr),
           colocate_gradients_with_ops=True)
 
-      return x, total_loss, train_op
+      return tf.contrib.learn.ModelFnOps(
+          mode=mode, predictions=x, loss=total_loss, train_op=train_op)
+
```
Comment (aidangomez) [inline models/deep_ae.py]: Don't need two spaces here, again PyYapf will fix this

File: models/deep_ae.py [hunk 7]
```diff
@@ -36,14 +37,15 @@ def autoencoder(features, labels, mode):
           kernel_initializer=tf.random_normal_initializer())
 
       # TODO(aidan): generalize the logging
-      tf.summary.image("reconstr", tf.reshape(x,
-                                              [params.batch_size, 28, 28, 1]))
+      tf.summary.image("reconstr", tf.reshape(x, [params.batch_size, 28, 28,
+                                                  1]))
```
Comment (aidangomez) [inline models/deep_ae.py]: not necessary, PyYapf

File: hparams/registry.py [hunk 8]
```diff
@@ -9,7 +9,9 @@
     encoder_depth=2,
     decoder_depth=2,
     hidden_size=100,
-    is_copy=True)
+    is_copy=True,
+    label_smoothing=0.1)
+
```
Comment (aidangomez) [inline hparams/registry.py]: PyYapf

File: data/mnist.py [hunk 9]
```diff
@@ -41,5 +43,30 @@ def _input_fn():
...
+  data_set = mnist.train if training else mnist.test
+
+  def _input_fn():
+    input_images = tf.constant(data_set.images)
+
+    input_labels = tf.constant(
+        data_set.labels) if not params.is_copy else tf.constant(
+            data_set.images)
+
+    image, label = tf.train.slice_input_producer([input_images, input_labels])
```
Comment (aidangomez) [inline data/mnist.py]: PyYapf

File: models/deep_ae.py (hunk 7 above)
Comment (1vn) [inline models/deep_ae.py]: weird, PyYapf is forcing this format even when I manually change it back

**Pull request merged**